translator/internaldata/resource_to_oc.go

```diff
@@ -106,17 +107,48 @@ func internalResourceToOC(resource pdata.Resource) (*occommon.Node, *ocresource.
 	return &ocNode, &ocResource
 }
 
-func attributeValueToString(attr pdata.AttributeValue) string {
+func attributeValueToString(attr pdata.AttributeValue, jsonLike bool) string {
 	switch attr.Type() {
+	case pdata.AttributeValueNULL:
+		if jsonLike {
+			return "null"
+		}
+		return ""
 	case pdata.AttributeValueSTRING:
+		if jsonLike {
+			return fmt.Sprintf("%q", attr.StringVal())
+		}
 		return attr.StringVal()
+
 	case pdata.AttributeValueBOOL:
 		return strconv.FormatBool(attr.BoolVal())
+
 	case pdata.AttributeValueDOUBLE:
 		return strconv.FormatFloat(attr.DoubleVal(), 'f', -1, 64)
+
 	case pdata.AttributeValueINT:
 		return strconv.FormatInt(attr.IntVal(), 10)
+
+	case pdata.AttributeValueMAP:
+		// OpenCensus attributes cannot represent maps natively. Convert the
+		// map to a JSON-like string.
```

Comment:
Why not JSON? Do we want this to be just human-readable representation or do we expect someone to attempt to translate it back?

Reply:
It is a stringified JSON. I think it is a good compromise since it preserves the full information and can be translated back, yet it is also more or less human readable.
I am open to other suggestions on how else we can translate this.